src/group_theory/free_abelian_group_finsupp.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import group_theory.free_abelian_group
+import data.finsupp.basic
+
+/-!
+# Isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+
+In this file we construct the canonical isomorphism between `free_abelian_group X` and `X →₀ ℤ`.
+We use this to transport the notion of `support` from `finsupp` to `free_abelian_group`.
+
+## Main declarations
+
+- `free_abelian_group.equiv_finsupp`: group isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+- `free_abelian_group.coeff`: the multiplicity of `x : X` in `a : free_abelian_group X`
+- `free_abelian_group.support`: the finset of `x : X` that occur in `a : free_abelian_group X`
+
+-/
+
+noncomputable theory
+
+open_locale big_operators
+
+variables (X : Type*)
+
+/-- The group homomorphism `free_abelian_group X →+ (X →₀ ℤ)`. -/
+def free_abelian_group.to_finsupp : free_abelian_group X →+ (X →₀ ℤ) :=
+free_abelian_group.lift $ λ x, finsupp.single x (1 : ℤ)
```

Comment:
I wonder if it would not be interesting to let `1 : int` as an implicit variable, rather than defyning "the" canoncail isomorphism.  This would allow to say something about subgroups of `free_abelian_group X` being of the form `free_abelian_group Y` for some `Y < X'` and some `X'` such that `free_abelian_group X'=free_abelian_group X'`. This would allow to combine your results with `fisupp.emb_domain` to extend the correspondance. On the other hand, with the actual API, the subgroup obtained by sending all `x : X` to `2 : int` is not reachable (I guess).

Reply:
I think the main focus of this PR is to build an API around the equivalence between `free_abelian_group` and `finsupp`. So, I think we can revisit this comment if/when it comes up in the future.